# Cобственное исключение
class MyException(Exception):
    pass


# Функция, которая вызывает исключение, если число не положительно
def positive_number(n):
    try:
        if n <= 0:  # Проверка, является ли число положительным
            raise MyException(f'{n} не является положительным числом')
        print(f'Введено положительное число: {n}')  # Вывод положительного числа
    except MyException as e:
        print(f'Поймано исключение: {e}')  # Вывод информации об исключении


# Функция, которая вызывает исключение, если строка не содержит только цифры
def only_digits(s):
    try:
        if not s.isalpha():  # Проверка, содержит ли строка только буквы
            raise MyException(f'{s} содержит не только буквы')
        print(f'Введена строка только из букв: {s}')  # Вывод строки, состоящей только из букв
    except MyException as e:
        print(f'Поймано исключение: {e}')  # Вывод информации об исключении


# Тестирование функций
positive_number(-10)
positive_number(11)
only_digits('квак')
only_digits('квак!')
